irodsConnector/session.py

```diff
@@ -170,90 +201,30 @@ def password(self) -> str:
-    def password(self):
-        """iRODS password deleter method.
-
-        """
-        self._password = ''
+        irods_auth_file = os.environ.get(
+            'IRODS_AUTHENTICATION_FILE', None)
+        if irods_auth_file is None:
+            irods_auth_file = utils.path.LocalPath('~/.irods/.irodsA').expanduser()
+        if irods_auth_file.exists():
+            with open(irods_auth_file, encoding='utf-8') as authfd:
+                self._password = irods.password_obfuscation.decode(authfd.read())
+                return self._password
 
     @property
-    def session(self) -> irods.session.iRODSSession:
-        """iRODS session creation.
+    def irods_session(self):
+        if self._irods_session:
+            return self._irods_session
+        return None
 
-        Returns
-        -------
-        iRODSSession
-            iRODS connection based on given environment and password.
-
-        """
-        if self._session is None:
-            options = {
-                'irods_env_file': str(self.context.irods_env_file),
-            }
-            if self.ienv is not None:
-                options.update(self.ienv)
-            given_pass = self.password
-            del self.password
-            # Accessing reset password property scrapes cached password.
-            cached_pass = self.password
-            del self.password
-            if given_pass != cached_pass:
-                options['password'] = given_pass
-            self._session = self._get_irods_session(options)
-            # If session exists, it is validated.
-            if self._session:
-                if given_pass != cached_pass:
-                    self._write_pam_password()
-                print('Welcome to iRODS:')
-                print(f'iRODS Zone: {self._session.zone}')
-                print(f'You are: {self._session.username}')
-                print(f'Default resource: {self.default_resc}')
-                print('You have access to: \n')
-                home_path = f'/{self._session.zone}/home'
-                if self._session.collections.exists(home_path):
-                    colls = self._session.collections.get(home_path).subcollections
-                    print('\n'.join([coll.path for coll in colls]))
-                logging.info(
-                    'IRODS LOGIN SUCCESS: %s, %s, %s', self._session.username,
-                    self._session.zone, self._session.host)
-        return self._session
-
-    @session.deleter
-    def session(self):
+    def del_irods_session(self):
         """Properly delete irods session.
         """
-        if self._session is not None:
-            self._session.cleanup()
-            del self._session
-            self._session = None
+        if self._irods_session is not None:
+            self._irods_session.cleanup()
+            del self._irods_session
+            self._irods_session = None
 
-    @staticmethod
-    def _get_irods_session(options):
+    def _connect_to_irods(self):
```

Comment:
I don't understand the need to _manually_ manage the attached `iRODSSession` with separate logical methods.  Simply accessing, assigning, or deleting a `@property` performs these operations automatically.  If a **session** does not exist, the **getter** initiates one, if one does exist, it bypasses that process and simply returns it.  That is the _purpose_ of a `@property`.

I can see that an inadvertent access of a property could cause trouble.  Using a method to check the private shadow variable can always be provided to check its status.  So far, I have found this unnecessary on the whole, but while writing unit tests recently, I see some possible potential need.

Ultimately, it is simplest to allow the object to do the work _for_ you.  Ask it what its state is if you're curious, but let it do what is needed based on that state.  The PRC uses quite a number of such `@property`s, the greatest number in the `session` module, in fact.  I don't see a reason _not_ to use them similarly here when they are beneficial.